### ISSUE TITLE
Switching to WebView doesn't work correctly in Appium mode when there is more than one WebView

### ISSUE THREAD

**Issue by pvlasov** [appium/appium#5903]
I have a hybrid app with two activities each with a web view. When I use Appium mode switching the context to the WebView always uses the first WebView, even when it is not visible anymore (I used getCurrentUrl() and getPageSource() to find it out). With Selendroid mode context switching works fine - it uses the currently visible WebView. However, screenshots don't work in Selendroid mode, although both context switching and screenshot taking work perfectly fine in pure Selendroid.

**Comment by liu3821** [comment]
i have same issue, and have resolved it .
1.print all web view, then chose the one you want.
2. and a web view may include many windows(like a browser open multiple web site), so you need switch window.
hope this can help you, good luck.

Java:
// print all context
Set<String> contextNames = ad.getContextHandles();
for (String contextName : contextNames) {
  System.out.println(contextName);
}

// switch window(find Facebook )
 Object[] windowHandles = ad.getWindowHandles().toArray();
	    for(int i=0; i<windowHandles.length; i++){
	    	ad.switchTo().window(windowHandles[i].toString());
	    	
	    	System.out.println("webview getTitle:" + ad.getTitle());
	    	
	    	if(ad.getTitle().equals("Welcome to Facebook")){
	    		break;
	    	}
	    }

**Comment by pvlasov** [comment]
I tried to print all contexts in each activity - the list is the same - NATIV_APP and WebView_<package name> in Appium and WebView_0 in Selendroid. However, Selendroid understands that WebView_0 in Activity A is different from WebView_0 in Activity B and switches to the currently visible/active WebView, but Appium always uses the first WebView it's encountered, even if it is not visible/active anymore.

I see your point that maybe both webviews in different activities are backed by a single webview with multiple windows and this is why in addition to switching contexts I have to switch windows. I'd try your approach, but I've already switched my tests to Selendroid. I'll keep it in mind if I run into this problem again.

I think that your approach may be considered a workaround, but the issue still stands because a) Appium behavior is different from Selendroid b) I have two distinct web views and I expect to be able to switch between them using ContextAware methods - having to switch between windows in addition to switching between contexts is kind of not obvious.
